rosidl_typesupport_opendds_cpp/resource/idl__rosidl_typesupport_opendds_cpp.idl.em

```diff
@@ -0,0 +1,85 @@
+// generated from rosidl_typesupport_opendds_cpp/resource/idl__rosidl_typesupport_opendds_cpp.idl.em
+// with input from @(package_name):@(interface_path)
+// generated code does not contain a copyright notice
+
+@{
+#######################################################################
+# EmPy template for generating <idl>__rosidl_typesupport_opendds_cpp.idl files
+#
+# Context:
+#  - package_name (string)
+#  - content (rosidl_parser.definition.IdlContent result of parsing IDL file)
+#  - interface_path (Path relative to the directory named after the package)
+#######################################################################
+from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+include_parts = [package_name] + list(interface_path.parents[0].parts) + \
+    [convert_camel_case_to_lower_case_underscore(interface_path.stem)]
+header_guard_variable = '__'.join([x.upper() for x in include_parts]) + \
+    '__ROSIDL_TYPESUPPORT_OPENDDS_CPP_HPP_'
+include_directives = set()
+}@
+
+#ifndef @(header_guard_variable)
+#define @(header_guard_variable)
+
+@{from rosidl_parser.definition import Service}
+@[for service in content.get_elements_of_type(Service)]
+#include "@(service.namespaced_type.name)_.idl"
+#include "dds/DdsDcpsGuid.idl"
+#include "dds/DCPS/RTPS/RtpsCore.idl"
+
+module typesupport_opendds_cpp {
+struct SampleIdentity {
+  OpenDDS::DCPS::GUID_t writer_guid;
+  OpenDDS::RTPS::SequenceNumber_t sequence_number;
+}; 
+}; // module typesupport_opendds_cpp
```

Comment:
Very good question. I've thought about this too. I was trying to make this class less RMW specific, and more RPC generic. Not sure how this benefit would outweigh added implementation complexity.

Reply:
If we're reusing an existing struct from RMW, I think it would be ok. I assume that we'd be ok with even pulling some of the RMW structures (possibly by a different name) into an OpenDDS implementation of RPC if it makes sense.

Reply:
I agree that we should use the rmw struct here. The future OpenDDS RPC is not likely to use type support to do all these. Also, there are sufficient GUID, RepoId, and functions in OpenDDS to handle id's and seq#. Defining a new type complicates things here in rmw, and may not benefit OpenDDS in the future, either.

Reply:
We probably should keep it as RPC compatible type though. RMW types should not 'penetrate' into the RPC layer, if possible. Even if the future OpenDDS RPC will not use RMW type support, it is RMW OpenDDS and its type support that could use Open DDS RPC, and its API and logic would not have to change. Current (OMG) type provides better portability of Requester / Replier classes and better compatibility of RMW OpenDDS with OMG RPC spec. Also, with the  recent changes I've made to the code, the negative impact of this approach is minimal. Having said that, I remain open to the change, but would like to hear Adam's opinion too.

Reply:
As you might have guessed, I'd try to align it as much with the OMG spec as possible.  But I have a different question... why is this in an em template if it's not specific to ROSIDL types?  If it's common, move it to regular idl file, not one generated by em.

Reply:
Put RPC types in RPC.idl